### PR TITLE
fix(parser): fix parsing nested wildcard struct field access

### LINKED ISSUE #7011: nested wildcard struct field access panics when there are additional parentheses
### Describe the bug

`SELECT (((((1,2,3)::foo).v1))).*`

test case in sqlparser 

https://github.com/risingwavelabs/risingwave/blob/2f4f7cb7230803759b7bc92626d7db5c4b2e195b/src/sqlparser/tests/testdata/select.yaml#L42-L43

 panicked at 'internal error: entered unreachable code', src/frontend/src/binder/struct_field.rs:81:22

<details>
<summary>AST</summary> <code>
Query(
    Query {
        with: None,
        body: Select(
            Select {
                distinct: All,
                projection: [
                    ExprQualifiedWildcard(
                        FieldIdentifier(
                            Cast {
                                expr: Row(
                                    [
                                        Value(
                                            Number(
                                                "1",
                                            ),
                                        ),
                                        Value(
                                            Number(
                                                "2",
                                            ),
                                        ),
                                        Value(
                                            Number(
                                                "3",
                                            ),
                                        ),
                                    ],
                                ),
                                data_type: Custom(
                                    ObjectName(
                                        [
                                            Ident {
                                                value: "foo",
                                                quote_style: None,
                                            },
                                        ],
                                    ),
                                ),
                            },
                            [
                                Ident {
                                    value: "v1",
                                    quote_style: None,
                                },
                            ],
                        ),
                        None,
                    ),
                ],
                from: [],
                lateral_views: [],
                selection: None,
                group_by: [],
                having: None,
            },
        ),
        order_by: [],
        limit: None,
        offset: None,
        fetch: None,
    },
)
</code>
</details>

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
Fix #7011: nested wildcard struct field access panics when there are additional parentheses.

Also did some minor style refactoring and added some comments.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/frontend/planner_test/tests/testdata/subquery.yaml [hunk 1]
```diff
@@ -182,7 +182,7 @@
     └─LogicalApply { type: LeftOuter, on: true, correlated_id: 1, max_one_row: true }
       ├─LogicalFilter { predicate: In(pg_class.relkind, 'r':Varchar, 'p':Varchar, 'v':Varchar, 'm':Varchar, 'S':Varchar, 'f':Varchar, '':Varchar) AND (pg_namespace.nspname <> 'pg_catalog':Varchar) AND IsNull(RegexpMatch(pg_namespace.nspname, '^pg_toast':Varchar)) AND (pg_namespace.nspname <> 'information_schema':Varchar) }
       | └─LogicalJoin { type: LeftOuter, on: (pg_namespace.oid = pg_class.relnamespace), output: all }
-      |   ├─LogicalScan { table: pg_class, columns: [pg_class.oid, pg_class.relname, pg_class.relnamespace, pg_class.relowner, pg_class.relkind] }
+      |   ├─LogicalScan { table: pg_class, columns: [pg_class.oid, pg_class.relname, pg_class.relnamespace, pg_class.relowner, pg_class.relkind, pg_class.relam, pg_class.reltablespace] }
```
Comment (xxchan) [inline src/frontend/planner_test/tests/testdata/subquery.yaml:185]: This is not introduced in this PR. An clean run will also produce it.

File: src/sqlparser/src/parser.rs [hunk 2]
```diff
@@ -268,47 +277,64 @@ impl Parser {
...
-            } else {
-                Ok(WildcardOrExpr::ExprQualifiedWildcard(expr, id_parts))
-            }
-        } else if let WildcardOrExpr::Wildcard = wildcard_expr {
-            Ok(WildcardOrExpr::ExprQualifiedWildcard(expr, vec![]))
-        } else {
-            Ok(wildcard_expr)
+        if let WildcardOrExpr::Expr(e) = simple_wildcard_expr {
+            return Ok(WildcardOrExpr::Expr(e));
+        }
```
Comment (xxchan) [inline src/sqlparser/src/parser.rs:304]: Actually I didn't fully understand the `WildcardOrExpr::Expr` stuff. But it works.... 😇🤡

File: src/sqlparser/src/parser.rs [hunk 3]
```diff
@@ -246,17 +255,17 @@ impl Parser {
             Token::Mul => {
                 return Ok(WildcardOrExpr::Wildcard);
             }
-            // TODO: support (((table.v1).v2).*)
```
Comment (xxchan) [inline src/sqlparser/src/parser.rs:249]: This TODO caused the panic

**Comment by st1page** [comment]
Yes, PG support that...
```
dev=# CREATE TYPE my_type_a AS (
    a int      
);
CREATE TYPE
dev=# CREATE TYPE my_type_b_my_type_a AS (
    b my_type_a
);
CREATE TYPE
dev=# create table t(c my_type_b_my_type_a); 
CREATE TABLE
dev=# select ((t.c).b).a from t;
 a 
---
(0 rows)

dev=# select ((t.c).b).* from t;
 a 
---
(0 rows)

```

File: src/frontend/planner_test/tests/testdata/subquery.yaml (hunk 1 above)
Comment (st1page) [reply src/frontend/planner_test/tests/testdata/subquery.yaml:185]: @yezizp2012 Any idea? :confused:

File: src/sqlparser/src/parser.rs (hunk 3 above)
Comment (st1page) [reply src/sqlparser/src/parser.rs:249]: Yes, PG support that...
```
dev=# CREATE TYPE my_type_a AS (
    a int      
);
CREATE TYPE
dev=# CREATE TYPE my_type_b_my_type_a AS (
    b my_type_a
);
CREATE TYPE
dev=# create table t(c my_type_b_my_type_a); 
CREATE TABLE
dev=# select ((t.c).b).a from t;
 a 
---
(0 rows)

dev=# select ((t.c).b).* from t;
 a 
---
(0 rows)

dev=# select (((t.c).b).*) from t;
 a 
---
(0 rows)

```

**Review by st1page — APPROVED** [review]
generally LGTM.

File: src/frontend/planner_test/tests/testdata/subquery.yaml (hunk 1 above)
Comment (yezizp2012) [reply src/frontend/planner_test/tests/testdata/subquery.yaml:185]: The two columns were introduced in #7008. I have no idea why the test passed. 🥵

**Review by yezizp2012 — APPROVED** [review]

File: src/frontend/planner_test/tests/testdata/subquery.yaml (hunk 1 above)
Comment (xxchan) [reply src/frontend/planner_test/tests/testdata/subquery.yaml]: https://buildkite.com/risingwavelabs/pull-request/builds/14347#01853502-2d2f-48b0-ae29-be5e2a56cc6c test failed but ci passed

**Pull request merged**